Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/Pages/PagesControllerImpl.cs

```diff
@@ -1,1485 +1,1503 @@
+                    allTerms.AddRange(termController.GetTermsByVocabulary(v.VocabularyId));
+                }
+
+                foreach (var tag in pageSettings.Tags.Trim().Split(','))
+                {
+                    if (!string.IsNullOrEmpty(tag) && tab.Terms.All(t => !t.Name.Equals(tag, StringComparison.OrdinalIgnoreCase)))
+                    {
+                        var term = allTerms.FirstOrDefault(t => t.Name.Equals(tag, StringComparison.OrdinalIgnoreCase));
+                        if (term == null)
+                        {
+                            var termId = termController.AddTerm(new Term(tag, string.Empty, vocabularyId));
+                            term = termController.GetTerm(termId);
+                        }
+
+                        tab.Terms.Add(term);
+                    }
+                }
+            }
+
+            // icons
+            if (pageSettings.IconFile != null && pageSettings.IconFile.fileId > 0)
+            {
+                tab.IconFile = FileManager.Instance.GetFile(pageSettings.IconFile.fileId).RelativePath;
+            }
+            else
+            {
+                tab.IconFile = null;
+            }
+            if (pageSettings.IconFileLarge != null && pageSettings.IconFileLarge.fileId > 0)
+            {
+                tab.IconFileLarge = FileManager.Instance.GetFile(pageSettings.IconFileLarge.fileId).RelativePath;
+            }
+            else
+            {
+                tab.IconFileLarge = null;
+            }
+
+        }
+
+        protected IOrderedEnumerable<KeyValuePair<int, string>> GetLocales(int portalId)
+        {
+            var locales = new Lazy<Dictionary<string, Locale>>(() => LocaleController.Instance.GetLocales(portalId));
+            return locales.Value.Values.Select(local => new KeyValuePair<int, string>(local.KeyID, local.EnglishName)).OrderBy(x => x.Value);
+        }
+
+        protected IEnumerable<KeyValuePair<int, string>> GetSiteAliases(int portalId)
+        {
+            var aliases = PortalAliasController.Instance.GetPortalAliasesByPortalId(portalId);
+            return aliases.Select(alias => new KeyValuePair<int, string>(alias.KeyID, alias.HTTPAlias)).OrderBy(x => x.Value);
+        }
+
+        protected int? GetPrimaryAliasId(int portalId, string cultureCode)
+        {
+            var aliases = PortalAliasController.Instance.GetPortalAliasesByPortalId(portalId);
+            var primary = aliases.Where(a => a.IsPrimary
+                                             && (a.CultureCode == cultureCode || String.IsNullOrEmpty(a.CultureCode)))
+                .OrderByDescending(a => a.CultureCode)
+                .FirstOrDefault();
+            return primary == null ? (int?)null : primary.KeyID;
+        }
+
+        private static string GetExternalUrlRedirection(string url)
+        {
+            if (url == null)
+            {
+                return null;
+            }
+
+            if (url.ToLower() == "http://")

```

Comment:
If we are making changes here, should we possibly consider defaulting to https:// as the default?

Reply:
No, this is checking to see if "http://" was injected in due to old data.  This is not setting anything.